tools/migrate_gid_to_uid/migrate_gid_to_uid.go

```diff
@@ -272,28 +296,60 @@ func buildGroupIDToUserIDMap(logger log.Logger, config *Config, metaIDToUserIDMa
 		Value  string `bson:"value"`
 	}
 	for iter.Next(&result) {
-		if metaID := result.MetaID; metaID == "" {
-			logger.Warn("Missing meta id in result from meta query")
-		} else if result.Value == "" {
-			logger.WithField("metaId", metaID).Warn("Missing value in result from meta query for meta id")
-		} else {
-			var value struct {
-				Private struct {
-					Uploads struct {
-						ID string `json:"id"`
-					} `json:"uploads"`
-				} `json:"private"`
-			}
-			if err = json.Unmarshal([]byte(result.Value), &value); err != nil {
-				logger.WithField("metaId", metaID).WithError(err).Warn("Unable to unmarshal value from meta query for meta id")
-			} else if groupID := value.Private.Uploads.ID; groupID == "" {
-				logger.WithField("metaId", metaID).Debug("Missing private uploads id in value in result from meta query for meta id")
-			} else if userID, ok := metaIDToUserIDMap[metaID]; !ok {
-				logger.WithField("metaId", metaID).Warn("Missing user id for meta id")
-			} else {
-				groupIDToUserIDMap[groupID] = userID
-			}
+		metaLogger := logger
+
+		metaID := result.MetaID
+		if metaID == "" {
+			metaLogger.Warn("Missing meta id in result from meta query")
+			continue
+		}
+
+		metaLogger = metaLogger.WithField("metaId", metaID)
+
+		if _, ok := metaIDMap[metaID]; ok {
+			metaLogger.Error("Found multiple metas with same meta id")
+			continue
 		}
+		metaIDMap[metaID] = true
+
+		userID, ok := metaIDToUserIDMap[metaID]
+		if !ok {
+			metaLogger.Error("Missing user id for meta id")
+			continue
+		}
+
+		metaLogger = metaLogger.WithField("userId", userID)
+
+		if result.Value == "" {
+			metaLogger.Warn("Missing value in result from meta query for meta id")
+			continue
+		}
+
+		var value struct {
+			Private struct {
+				Uploads struct {
+					ID string `json:"id"`
+				} `json:"uploads"`
+			} `json:"private"`
+		}
+		if err = json.Unmarshal([]byte(result.Value), &value); err != nil {
+			metaLogger.WithError(err).Warn("Unable to unmarshal value from meta query for meta id")
```

Comment:
It is a warning simply because there is already "bogus" data in our deployed databases (presumably from earlier incarnations of the code). So, overall in this script I use `Error` to indicate that something has failed and the script wasn't able to do its work and needs to be resolved, while `Warn` is for when something is not right, but can happen and not something that needs to be resolved.